src/Core/Entity/Image/InputImage.php

```diff
@@ -38,6 +42,33 @@ public function __construct(OptionsBag $optionsBag, string $sourceImageUrl)
 
         $this->sourceImagePath = $optionsBag->hashOriginalImageUrl($this->sourceImageUrl);
         $this->saveToTemporaryFile();
+
+        // Store the source file mime type.
+        $this->sourceFileMimeType = finfo_file(
```

Comment:
Mime type is handled by the ImageMetaInfo class. If the name would be misleading (since it's video) you could rename it to FileMetaInfo or MediaMetaInfo.

Reply:
Changed to use `ImageMetaInfo `